### PR TITLE
addresses issues/49

### PR DESCRIPTION
**Pull request by matthewpeterkort**
see https://github.com/ACED-IDP/gen3_util/issues/49

### CODE REVIEW

File: gen3_util/common/__init__.py [hunk 1]
```diff
@@ -23,6 +23,30 @@
...
+
+1. **Read-Only Access:** Unless you have explicit permission to modify or update the data, treat this directory as read-only.
+
+2. **Data Integrity:** Ensure the integrity of the data by following proper procedures for reading, updating, and managing files.
+
+3. **Security Awareness:** Be aware of the sensitivity of the data stored here and take necessary precautions to protect it from unauthorized access.
+
+## How to Obtain Access:
+
+If you need access to these files, please contact the project administrator for access to idp.cbds.ohsu.edu
```
Comment (bwalsh) [inline gen3_util/common/__init__.py]: My bad, can you remove these lines?  Thanks.

**Review by bwalsh — CHANGES_REQUESTED** [review]
Can you modify the unit test  `tests.integration.test_init.test_init`  to verify these files were created?  Thank You.

File: gen3_util/common/__init__.py (hunk 1 above)
Comment (matthewpeterkort) [reply gen3_util/common/__init__.py]: see https://github.com/ACED-IDP/gen3_util/pull/63/commits/4ce7639c66878522324603955f7792822cd83417

**Comment by matthewpeterkort** [comment]
> Can you modify the unit test `tests.integration.test_init.test_init` to verify these files were created? Thank You.

see https://github.com/ACED-IDP/gen3_util/pull/63/commits/4ce7639c66878522324603955f7792822cd83417

File: gen3_util/config/__init__.py [hunk 2]
```diff
@@ -247,6 +247,21 @@ def init(config: Config, project_id: str) -> Generator[str, None, None]:
...
+    if not pathlib.Path(state_dir_git_ignore).exists():
+        with open(state_dir_git_ignore, 'w') as f:
+            f.write("*\n!README.md")
+
+    meta_dir_git_ignore = "META/.gitignore"
+    if not pathlib.Path(meta_dir_git_ignore).exists():
+        with open(meta_dir_git_ignore, 'w') as f:
+            f.write("*\n!README.md")
+
+    project_readme = PROJECT_DIR + '/README.md'
```
Comment (bwalsh) [inline gen3_util/config/__init__.py]: My notes in the issue were imprecise.

image

The intent was to create .g3t/README.md and META/README.md both with the warning not to commit files.

**Review by bwalsh — CHANGES_REQUESTED** [review]
Small change I missed before.

**Comment by matthewpeterkort** [comment]
> Can you modify the unit test `tests.integration.test_init.test_init` to verify these files were created? Thank You.

addressed in https://github.com/ACED-IDP/gen3_util/pull/63/commits/18215c9919a7bb7ebef436e7b1ab8b1cdd0625fd

**Comment by bwalsh** [comment]
@matthewpeterkort can you run:
* `pytest tests/unit`
* `pytest tests/integration`

I had to make the following changes to run these tests locally.  I did not commit the changes.

```diff --git a/gen3_util/common/__init__.py b/gen3_util/common/__init__.py
index 756ad45..f370585 100644
--- a/gen3_util/common/__init__.py
+++ b/gen3_util/common/__init__.py
@@ -30,7 +30,9 @@ Welcome to the data directory! This repository contains important data files for
 
 ## Important Note: Do Not Check in Protected Files
 
-Some files in this directory are considered protected and contain sensitive information. **DO NOT** check in or commit these protected files to the version control system (e.g., Git). This is crucial to prevent unauthorized access and to comply with security and privacy policies.
+Some files in this directory are considered protected and contain sensitive information.
+**DO NOT** check in or commit these protected files to the version control system (e.g., Git).
+This is crucial to prevent unauthorized access and to comply with security and privacy policies.
 
 
 ## Usage Guidelines:
diff --git a/tests/integration/test_commit.py b/tests/integration/test_commit.py
index a040121..5f5a5c2 100644
--- a/tests/integration/test_commit.py
+++ b/tests/integration/test_commit.py
@@ -57,7 +57,7 @@ def test_init_project(config, program, tmp_path, profile):
 
     metadata_path = pathlib.Path(tmp_path) / 'META'
     metadata_ls = sorted(str(_.relative_to(tmp_path)) for _ in metadata_path.glob('**/*.*'))
-    expected_metadata_ls = ['META/DocumentReference.ndjson', 'META/Patient.ndjson', 'META/ResearchStudy.ndjson', 'META/ResearchSubject.ndjson']
+    expected_metadata_ls = sorted(['META/.gitignore', 'META/README.md', 'META/DocumentReference.ndjson', 'META/Patient.ndjson', 'META/ResearchStudy.ndjson', 'META/ResearchSubject.ndjson'])
     assert metadata_ls == expected_metadata_ls, f"expected metadata files not found {metadata_ls}"
 
     result = runner.invoke(cli, f'--format json  --profile {profile} commit -m "test-commit"'.split())
@@ -123,7 +123,7 @@ def add_commit_push(runner: CliRunner, tmp_path, profile, project_id):
 
     metadata_path = pathlib.Path(tmp_path) / 'META'
     metadata_ls = sorted(str(_.relative_to(tmp_path)) for _ in metadata_path.glob('**/*.*'))
-    expected_metadata_ls = ['META/DocumentReference.ndjson', 'META/Patient.ndjson', 'META/ResearchStudy.ndjson', 'META/ResearchSubject.ndjson']
+    expected_metadata_ls = sorted(['META/.gitignore', 'META/README.md',  'META/DocumentReference.ndjson', 'META/Patient.ndjson', 'META/ResearchStudy.ndjson', 'META/ResearchSubject.ndjson'])
     assert metadata_ls == expected_metadata_ls, f"expected metadata files not found {metadata_ls}"
 
     result = runner.invoke(cli, f'--format json  --profile {profile} commit -m "test-commit"'.split())
@@ -249,7 +249,7 @@ def test_bundle(config, program, tmp_path):
     study_metadata(config=config, project_id=project_id, output_path=metadata_path,
                    overwrite=False, source='manifest')
     metadata_ls = sorted(str(_.relative_to(tmp_path)) for _ in metadata_path.glob('**/*.*'))
-    expected_metadata_ls = ['META/DocumentReference.ndjson', 'META/Patient.ndjson', 'META/ResearchStudy.ndjson', 'META/ResearchSubject.ndjson']
+    expected_metadata_ls = sorted(['META/.gitignore', 'META/README.md', 'META/DocumentReference.ndjson', 'META/Patient.ndjson', 'META/ResearchStudy.ndjson', 'META/ResearchSubject.ndjson'])
     assert metadata_ls == expected_metadata_ls, f"expected metadata files not found {metadata_ls}"
 
     # create a bundle
```

**Comment by bwalsh** [comment]
Otherwise LGTM:

```
g3t init test-gitignore
git init
echo test >  .g3t/state/foo
echo test > META/Patient.ndjson
git add .
>>>
Changes to be committed:
  (use "git rm --cached <file>..." to unstage)
        new file:   .g3t/README.md
        new file:   .g3t/config.yaml
        new file:   META/README.md

```

**Comment by matthewpeterkort** [comment]
> @matthewpeterkort can you run:

see https://github.com/ACED-IDP/gen3_util/pull/63/commits/78a855b0c21148d9f4459f50570cdb6c5b74c84e

**Review by bwalsh — APPROVED** [review]
LGTM

**Pull request merged**